views/components/ConfigurationCard/index.jsx

```diff
@@ -0,0 +1,187 @@
+import React from 'react'
+import moment from 'moment'
+import {
+  Card,
+  CardHeader,
+  Divider,
+  Typography,
+  CardActions,
+  Switch,
+  IconButton,
+  FormControlLabel,
+} from '@material-ui/core'
+import { Edit, Clear, Share } from '@material-ui/icons'
+import FullView from '@material-ui/icons/AspectRatio'
+import Copy from '@material-ui/icons/FileCopy'
+import ConfigCardAvatar from '../ConfigurationCardAvatar'
+import openNewWindow from '../../fe-utils/windowUtil'
+import { UserModel, UserConfigurationsModel } from '../../models'
+import { routes } from '../../routes/routes'
+import { colors } from '../../../constants'
+
+const ConfigurationCard = ({
+  classes,
+  openSearch,
+  loadAllConfigurations,
+  user,
+  config,
+  preferences,
+  setPreferences,
+  setConfigurations,
+  width,
+}) => {
+  const { uid } = user
+  const { _id, name, owner, readers, type } = config
+  const ownsConfig = uid === owner
+  const showTime = config.modified || config.created
+  const localTime = moment.utc(showTime).local().format()
+  const updated = moment(localTime).calendar()
+  const checked = config._id === preferences.config
+
+  const copyConfig = async (configuration) => {
+    const { _id, ...configAttributes } = configuration
+    const newConfig = {
+      ...configAttributes,
+      owner: uid,
+      readers: [uid],
+      created: new Date().toUTCString(),
+    }
+    const { status } = await UserConfigurationsModel.create(uid, newConfig)
+
+    if (status === 200) loadAllConfigurations(uid)
+  }
+
+  const removeConfig = async (configId) => {
+    const res = await UserConfigurationsModel.destroy(uid, configId)
+    if (res.status === 200) {
+      setConfigurations((configurations) =>
+        configurations.filter(({ _id }) => _id !== configId)
+      )
+
+      if (checked) updateUserPreferences(configId)
+    }
+  }
+
+  const updateConfiguration = async (configId, configAttributes) => {
+    const { status } = await UserConfigurationsModel.update(
+      uid,
+      configId,
+      configAttributes
+    )
+
+    switch (true) {
+      case status === 200: {
+        loadAllConfigurations(uid)
+        break
+      }
+      case checked: {
+        updateUserPreferences(configId)
+        break
+      }
+      default:
+        break
+    }
```

Comment:
why not a standard conditional here?

Reply:
[d31c716](https://github.com/AMP-SCZ/dpdash/pull/310/commits/d31c716b584e340b90068954fc940c49761e52fe)